### PR TITLE
Do not generate .deps on clean

### PR DESCRIPTION
**Pull request by Disasm**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,4 +100,6 @@ clean: clean-rs clean-patch clean-html
 	@mkdir -p .deps
 	python3 scripts/makedeps.py $< > $@
 
--include $(patsubst devices/%.yaml, .deps/%.d, $(YAMLS))
+$(PATCHED_SVDS): $(patsubst devices/%.yaml, .deps/%.d, $(YAMLS))
```
Comment (adamgreig) [inline Makefile]: Could we stick the `.deps/%.d` dependency on L37 with `svd/%.svd.patched` instead?

Comment (Disasm) [reply Makefile]: Hmm, this way make treats dep files as intermediate and removes them after build.

Comment (adamgreig) [reply Makefile]: I did a quick test and that seems to work great, ensuring each SVD only depends on its on `.d` file too.

Comment (adamgreig) [reply Makefile]: Ah, you're right about the intermediate business. Maybe stick them in .PRECIOUS too?

Comment (Disasm) [reply Makefile]: Fixed!

**Comment by adamgreig** [comment]
Thanks! Could you rebase off master, as it now contains `.PRECIOUS` for the `svd/%.svd` files as well.

**Comment by adamgreig** [comment]
Incidentally I think you could do `.PRECIOUS: .deps/%.d` rather than having to build the list.

**Comment by Disasm** [comment]
Yeah, I see. Didn't know about this.

**Comment by Disasm** [comment]
bors r=adamgreig

**Pull request merged**